main.cpp

```diff
@@ -1,17 +1,64 @@
 #include "main.h"
 #include "rc.h"
 #include <filesystem>
+#include <gdiplus.h>
 
-void main_window::on_paint(HDC hdc) 
-{
+void main_window::on_paint(HDC hdc) {
+	RECT rect;  // Lef, top, right, bottom
+	GetClientRect(*this, &rect);
+	Gdiplus::RectF layoutRect( rect.left, rect.top, rect.right, rect.bottom);  // X, Y, Width, Height
+	Gdiplus::Graphics graphics(hdc);
+
+	// Image
+	graphics.DrawImage(image, layoutRect);
+
+	if (image) {
+		Gdiplus::StringFormat stringFormat;
+		stringFormat.SetAlignment(Gdiplus::StringAlignmentCenter);
+		Gdiplus::Font font(_T("Arial"), 24, 1);  // style 1 = bold
+
+		// Shadow
+		Gdiplus::SolidBrush solidBrush(Gdiplus::ARGB(0xFF000000));  // Black
+
+		layoutRect.X = rect.left + 3;
+		layoutRect.Y = rect.bottom - 40;
+
+		graphics.DrawString(fileName, -1, &font, layoutRect, &stringFormat, &solidBrush);
+
+		// Text
+		layoutRect.X = rect.left;
+		layoutRect.Y = rect.bottom - 43;
+
+		solidBrush.SetColor(Gdiplus::ARGB(0xFFFFFFFF));  // White
+
+		graphics.DrawString(fileName, -1, &font, layoutRect, &stringFormat, &solidBrush);
+	}
+}
+
+bool main_window::on_erase_bkgnd(HDC hdc) {
+	return 1;
 }
 
 void main_window::on_command(int id) 
 {
 	switch (id) 
 	{
-		case ID_OPEN:
-			break;
+		case ID_OPEN: {
+			TCHAR path[MAX_PATH]; *path = 0;
+			TCHAR filter[] = _T("Image Files (*.gif;*.jpg;*.png;*.bmp;*emf)\0*.gif;*.jpg;*.jpeg;*.png;*.bmp;*.emf\0");
+			OPENFILENAME ofn{ sizeof OPENFILENAME };
+			ofn.hwndOwner = *this;
+			ofn.lpstrFile = path;
+			ofn.nMaxFile = MAX_PATH;
+			ofn.lpstrFilter = filter;
+			ofn.Flags = OFN_HIDEREADONLY;
+			if (::GetOpenFileName(&ofn)) {  // If specified file name and clicked on OK button: returns nonzero, OPENFILENAME contains full path and name.
+				image = Gdiplus::Image::FromFile(ofn.lpstrFile);
```

Comment:
A što ako je `image` već učitan? Niste obrisali prethodni.
A nemate ni destruktor da obrišete ovaj `image` (prijedlog: koristite `std::unique_ptr`).

Reply:
Što sve treba ručno brisati?